From 626ad3f353c9ce39b512ca4244f5eb2333bc8777 Mon Sep 17 00:00:00 2001
From: Zion Gal <galz@marvell.com>
Date: Mon, 1 Jun 2015 15:55:53 +0300
Subject: [PATCH 0186/1240] spi: added support for micron device N25Q1024

	- enabled CONFIG_SPI_FLASH_BAR to allow access to more than
	  lower 16MB
	- changed erase size for N25Q1024 from 4KB to 64KB. this chip
	  supports sector erase (64KB) and thus allows faster erase time

Change-Id: I64b76e4cbba405819406608644d301d529d11bf6
Signed-off-by: Zion Gal <galz@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/19703
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 drivers/mtd/spi/sf_params.c    | 2 +-
 include/configs/mvebu-common.h | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/spi/sf_params.c b/drivers/mtd/spi/sf_params.c
index c12e8c6..4306883 100644
--- a/drivers/mtd/spi/sf_params.c
+++ b/drivers/mtd/spi/sf_params.c
@@ -87,7 +87,7 @@ const struct spi_flash_params spi_flash_params_table[] = {
 	{"N25Q256A",	   0x20bb19, 0x0,       64 * 1024,   512, RD_FULL,	   WR_QPP | SECT_4K},
 	{"N25Q512",	   0x20ba20, 0x0,       64 * 1024,  1024, RD_FULL, WR_QPP | E_FSR | SECT_4K},
 	{"N25Q512A",	   0x20bb20, 0x0,       64 * 1024,  1024, RD_FULL, WR_QPP | E_FSR | SECT_4K},
-	{"N25Q1024",	   0x20ba21, 0x0,       64 * 1024,  2048, RD_FULL, WR_QPP | E_FSR | SECT_4K},
+	{"N25Q1024",	   0x20ba21, 0x0,       64 * 1024,  2048, RD_FULL, WR_QPP | E_FSR },
 	{"N25Q1024A",	   0x20bb21, 0x0,       64 * 1024,  2048, RD_FULL, WR_QPP | E_FSR | SECT_4K},
 #endif
 #ifdef CONFIG_SPI_FLASH_SST		/* SST */
diff --git a/include/configs/mvebu-common.h b/include/configs/mvebu-common.h
index 1b14a10..4f6d129 100644
--- a/include/configs/mvebu-common.h
+++ b/include/configs/mvebu-common.h
@@ -363,6 +363,7 @@
 	#define CONFIG_ENV_SPI_MAX_HZ           10000000        /*Max 50Mhz- will sattle on SPI bus max 41.5Mhz */
 	#define CONFIG_ENV_SPI_CS               0
 	#define CONFIG_ENV_SPI_BUS              0
+	#define CONFIG_SPI_FLASH_BAR
 
 	#ifndef CONFIG_SF_DEFAULT_SPEED
 	#define CONFIG_SF_DEFAULT_SPEED        1000000
-- 
1.9.1

